### DISCUSSION TITLE
Version 3.7 with arch x64 not found

### DISCUSSION THREAD

**Discussion by MoBosse** [open-sdg/open-sdg#2161] [Announcements]
> **When using Github to build your data repository, you may see failures with this "Version 3.7 with arch x64" error. There are two possible fixes:**
> * Upgrading Python: example [here](https://github.com/open-sdg/open-sdg-data-starter/pull/69/files)
> * Pinning version of Ubuntu: example [here](https://github.com/open-sdg/open-sdg-data-starter/pull/70/files)
>
> Original discussion below:

Hi all,
since last week the commits of one of my data repositories keep causing an error:

> Version 3.7 with arch x64 not found Available versions: 3.10.15 (x64) 3.11.11 (x64) 3.12.8 (x64) 3.9.20 (x64)

I already deleted all the content of the csv and md files to make sure there isn't anything left that could cause the error. 
Afterwards I copied the content of the other files from another repository where everything still works fine. No effect.

I also tried to change the python Version to some higher number (e.g. 3.9 or 3.12...) [here](https://github.com/SdgTestEnvironment/sdg-data/blob/develop/.github/workflows/deploy-to-staging.yml#L30-L33) and [here](https://github.com/SdgTestEnvironment/sdg-data/blob/develop/.github/workflows/test-pull-requests.yml#L12-L15) but it didn't work.

See failing build [here](https://github.com/SdgTestEnvironment/sdg-data/actions/runs/12350815623/job/34464453897)

Does anyone have any ideas as to why this could be?

**Comment by MoBosse** [comment]
@brockfanning I do not really understand what ubuntu has to do with it but may the described error has somthing to do with [this issue](https://github.com/actions/runner-images/issues/10636)?
If so, other users could be affected soon aswell...

**Reply by brockfanning** [reply]
@MoBosse Thanks for pointing this out - yes I've seen the same thing. It does seem like Github's "ubuntu-latest" machine no longer supports Python 3.7. I think there are a couple ways to mitigate this - one is by upgrading to Python 3.8 (or higher) and the other is to pin to "ubuntu-22.04". I've tried examples [here](https://github.com/open-sdg/open-sdg-data-starter/pull/69) and [here](https://github.com/open-sdg/open-sdg-data-starter/pull/70) and both of them seem to be working. I think we'll need to do an announcement about this, since you're right that this is going to affect most users of Open SDG.

**Reply by MoBosse** [reply]
Thanks a lot. Both fixes work!
